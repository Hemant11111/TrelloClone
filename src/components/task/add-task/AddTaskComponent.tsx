import React, { useCallback, useState } from "react";

import "./AddTask.scss";
import TaskModel from "../../../model/Task";
import { useDispatch } from "react-redux";
import { createTaskThunk } from "../../../redux/thunk/task.thunk";
import InputForm from "../../../shared-components/input-form/InputForm";

interface AddTaskComponentProps {
    cardId: string;
}

export default function AddTaskComponent(props: AddTaskComponentProps) {

    const dispatch = useDispatch();

    const [title, setTitle] = useState<string>("");

    const handleNewTaskValueChange = useCallback((event: any) => {
        setTitle(event.target.value);
    }, [setTitle]);

    const handleAddNewTask = useCallback(() => {
        const newTask = TaskModel.newInstance(title, props.cardId)
        setTitle("");

        dispatch(createTaskThunk(newTask));
    }, [title])


    const handleTaskCreate = useCallback((e: any) => {
        handleAddNewTask();
        e.preventDefault();
    }, [title]);

    return (
        <div className="add-task-container">
            <InputForm
                placeholder="Add Task..."
                value={title}
                onSubmit={handleTaskCreate}
                onChange={handleNewTaskValueChange}
            />
            <button
                onClick={handleAddNewTask}
                className="add-task-button trello-icon-button"
                disabled={!title}
            ><i className="fa fa-plus"/>
            </button>
        </div>
    )
}