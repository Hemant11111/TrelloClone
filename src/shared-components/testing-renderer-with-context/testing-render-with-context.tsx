import { ReactElement } from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../../redux/store/store";

export function testingRenderWithContext(element: ReactElement) {
    return render(
        <Provider store={store}>
            {element}
        </Provider>
    )
}