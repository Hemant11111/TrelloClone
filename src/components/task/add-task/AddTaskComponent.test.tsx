import React from 'react';
import { screen } from "@testing-library/react";

import AddTaskComponent from "./AddTaskComponent";
import { testingRenderWithContext } from "../../../shared-components/testing-renderer-with-context/testing-render-with-context";


describe('Add task Component', () => {
    it('should match snapshot', () => {
        const {asFragment} = testingRenderWithContext(<AddTaskComponent cardId={"1"}/>);
        expect(asFragment()).toMatchSnapshot();
    })

    it('should render add button', () => {
        testingRenderWithContext(<AddTaskComponent cardId={"1"}/>);
        const button = screen.getByTestId("add-task-button");
        expect(button).toBeInTheDocument();
    })

    it('should render input field', () => {
        testingRenderWithContext(<AddTaskComponent cardId={"1"}/>);
        const inputField = screen.getByPlaceholderText("Add Task...");
        expect(inputField).toBeInTheDocument();
    })
})