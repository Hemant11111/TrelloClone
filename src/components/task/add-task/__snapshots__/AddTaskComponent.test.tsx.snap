// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add task Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="add-task-container"
  >
    <form
      class="custom-form"
    >
      <input
        class=" input-form"
        placeholder="Add Task..."
        value=""
      />
    </form>
    <button
      class="add-task-button trello-icon-button"
      data-testid="add-task-button"
      disabled=""
    >
      <i
        class="fa fa-plus"
      />
    </button>
  </div>
</DocumentFragment>
`;
